### PR TITLE
Fix error code when creating/registering a key with invalid id

### PR DESCRIPTION
**Pull request by ronald-cron-arm**
## Description
Fix 4271

## Status
READY

## Requires Backporting
No PSA only

### CODE REVIEW

File: tests/suites/test_suite_psa_crypto.data [hunk 1]
```diff
@@ -892,34 +892,42 @@ copy_success:PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_DECRYPT | PSA_KEY_USAGE_COPY
...
+
+Copy fail: AES, invalid persistent key identifier in attributes
+depends_on:PSA_WANT_ALG_CTR:PSA_WANT_KEY_TYPE_AES:MBEDTLS_PSA_CRYPTO_STORAGE_C
+copy_fail:PSA_KEY_USAGE_COPY | PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_EXPORT:PSA_ALG_CTR:0:PSA_KEY_TYPE_AES:"404142434445464748494a4b4c4d4e4f":PSA_KEY_TYPE_AES:0:PSA_KEY_USAGE_COPY | PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_EXPORT:PSA_ALG_CTR:0:0:PSA_KEY_LIFETIME_PERSISTENT:PSA_ERROR_INVALID_ARGUMENT
+
+Copy fail: AES, invalid lifetime (unknown location) in attributes
+depends_on:PSA_WANT_ALG_CTR:PSA_WANT_KEY_TYPE_AES:MBEDTLS_PSA_CRYPTO_STORAGE_C
+copy_fail:PSA_KEY_USAGE_COPY | PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_EXPORT:PSA_ALG_CTR:0:PSA_KEY_TYPE_AES:"404142434445464748494a4b4c4d4e4f":PSA_KEY_TYPE_AES:0:PSA_KEY_USAGE_COPY | PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_EXPORT:PSA_ALG_CTR:0:0:PSA_KEY_LIFETIME_FROM_PERSISTENCE_AND_LOCATION(PSA_KEY_LIFETIME_PERSISTENT, 11):PSA_ERROR_INVALID_ARGUMENT
```
Comment (gilles-peskine-arm) [inline tests/suites/test_suite_psa_crypto.data:930]: This test case has an invalid id, so it's unconclusive.
```suggestion
copy_fail:PSA_KEY_USAGE_COPY | PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_EXPORT:PSA_ALG_CTR:0:PSA_KEY_TYPE_AES:"404142434445464748494a4b4c4d4e4f":PSA_KEY_TYPE_AES:0:PSA_KEY_USAGE_COPY | PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_EXPORT:PSA_ALG_CTR:0:1:PSA_KEY_LIFETIME_FROM_PERSISTENCE_AND_LOCATION(PSA_KEY_LIFETIME_PERSISTENT, 11):PSA_ERROR_INVALID_ARGUMENT
```

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]
There's a one-bit copypasta in a test case. Other than that LGTM

File: tests/suites/test_suite_psa_crypto.data (hunk 1 above)
Comment (ronald-cron-arm) [reply tests/suites/test_suite_psa_crypto.data]: Indeed, fixed as suggested.

Comment (gilles-peskine-arm) [reply tests/suites/test_suite_psa_crypto.data]: The test case now looks good to me, but the data is making `psa_copy_key` succeed, so there must be a validation bug that we missed.

Comment (gilles-peskine-arm) [reply tests/suites/test_suite_psa_crypto.data]: I filed https://github.com/ARMmbed/mbedtls/issues/4292 for the failure of the test case causing subsequent persistent key tests to fail.

Comment (ronald-cron-arm) [reply tests/suites/test_suite_psa_crypto.data]: Yes this revealed some interesting things about `psa_copy_key`, I am on it.

Comment (ronald-cron-arm) [reply tests/suites/test_suite_psa_crypto.data]: The location is supposed to be validated by the driver wrappers but as the copy through an opaque driver is not supported yet the validation was not done. I've added a check in `psa_copy_key` for the time being. The CI seems happy now.

**Review by gilles-peskine-arm — APPROVED** [review]

**Review by gabor-mezei-arm — APPROVED** [review]

**Pull request merged**